src/Form/FormAnalyzer.js

```diff
@@ -331,11 +345,16 @@ class FormAnalyzer {
         }
 
         // A form with many fields is unlikely to be a login form
-        const relevantFields = this.form.querySelectorAll(this.matching.cssSelector('genericTextField'));
+        const relevantFields = this.form.querySelectorAll(this.matching.cssSelector('genericTextInputField'));
         if (relevantFields.length >= 4) {
             this.increaseSignalBy(relevantFields.length * 1.5, 'many fields: it is probably not a login');
         }
 
+        // If we can't decide at this point, try reading password hints
+        if (this.areLoginOrSignupSignalsWeak()) {
```

Comment:
If we see more examples in triages, I might just remove this check. But for now feels safer.